DS and Algorithm/Day3/Selectionsort.py

```diff
@@ -0,0 +1,40 @@
+def selection_sort(arr):
+    """
+    Sorts an array using the Selection Sort algorithm.
+
+    Parameters:
+    - arr (list): The input list to be sorted in-place.
+
+    Returns:
+    None (The input list is sorted in-place).
+    """
+
+    # Get the length of the array
+    n = len(arr)
+
+    # Traverse the entire array
+    for i in range(n):
+        # Assume the current index is the minimum
+        min_index = i
+
+        # Iterate through the unsorted part of the array to find the minimum element
+        for j in range(i + 1, n):
+            # Compare the current element with the assumed minimum
+            if arr[j] < arr[min_index]:
+                # If the current element is smaller, update the minimum index
+                min_index = j
+
+        # Swap the found minimum element with the first element in the unsorted part
+        arr[i], arr[min_index] = arr[min_index], arr[i]
```

Comment:
**Optimization Issue**: The selection sort algorithm is implemented correctly but is inherently inefficient for large datasets due to its O(n^2) time complexity. <br> **Fix**: For large datasets, consider using more efficient sorting algorithms like mergesort or quicksort, which have better average and worst-case time complexities. <br> **Code Suggestion**: 
 ```
 # Suggestion to implement a more efficient sorting algorithm is conceptual and requires changing the algorithm choice.
 ```